numpy/testing/utils.py

```diff
@@ -509,7 +509,7 @@ def _build_err_msg():
             return
     except (NotImplementedError, TypeError):
         pass
-    if round(abs(desired - actual), decimal) != 0:
+    if around(abs(desired - actual), decimal) >= 10.0**(-decimal):
```

Comment:
I don't understand this logic. Either just round, or compare the absolute difference with some value.